packages/server/lib/socket-base.ts

```diff
@@ -48,18 +39,9 @@ const runnerEvents: RunnerEvent[] = [
   'paused',
   'test:after:hooks',
   'run:end',
-]
-
-type ReporterEvent =
-  'runner:restart'
-  | 'runner:abort'
-  | 'runner:console:log'
-  | 'runner:console:error'
-  | 'runner:show:snapshot'
-  | 'runner:hide:snapshot'
-  | 'reporter:restarted'
-
-const reporterEvents: ReporterEvent[] = [
+] as const
+
+const reporterEvents = [
```

Comment:
Would we want to leverage [freeze](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) here instead to persist to JS files, or is that too aggressive? As I am typing this I think the cast to `const` is probably enough 😆

Reply:
I'm not a types expert 😬 We've done this in other places so I took it as a opportunity to remove some redundancies

Reply:
I think the main difference here is that one would prevent types being pushed in at runtime vs just a TS check. Honestly don't think it matters too much either way